subsys/bluetooth/shell/iso.c

```diff
@@ -0,0 +1,256 @@
+
+	if (iso_chan.conn) {
+		printk("No channels available\n");
+		return -ENOMEM;
+	}
+
+	*chan = &iso_chan;
+
+	return 0;
+}
+
+struct bt_iso_server iso_server = {
+	.sec_level = BT_SECURITY_L1,
+	.accept = iso_accept,
+};
+
+static int cmd_listen(const struct shell *shell, size_t argc, char *argv[])
+{
+	int err;
+
+	if (argc > 1) {
+		iso_server.sec_level = *argv[1] - '0';
+	}
+
+	err = bt_iso_server_register(&iso_server);
+	if (err) {
+		shell_error(shell, "Unable to register ISO cap (err %d)",
+			    err);
+	}
+
+	return err;
+}
+
+static int cmd_bind(const struct shell *shell, size_t argc, char *argv[])
+{
+	struct bt_conn *conns[1];
+	struct bt_iso_chan *chans[1];
+	int err;
+
+	if (!default_conn) {
+		shell_error(shell, "Not connected");
+		return 0;
+	}
+
+	if (iso_chan.conn) {
+		shell_error(shell, "Already binded");
+		return 0;
+	}
+
+	conns[0] = default_conn;
+	chans[0] = &iso_chan;
+
+	if (argc > 1) {
+		chans[0]->qos->dir = strtol(argv[1], NULL, 0);
+	}
+
+	if (argc > 2) {
+		chans[0]->qos->interval = strtol(argv[2], NULL, 0);
+	}
+
+	if (argc > 3) {
+		chans[0]->qos->packing = strtol(argv[3], NULL, 0);
+	}
+
+	if (argc > 4) {
+		chans[0]->qos->framing = strtol(argv[4], NULL, 0);
+	}
+
+	if (argc > 5) {
+		chans[0]->qos->latency = strtol(argv[5], NULL, 0);
+	}
+
+	if (argc > 6) {
+		chans[0]->qos->sdu = strtol(argv[6], NULL, 0);
+	}
+
+	if (argc > 7) {
+		chans[0]->qos->phy = strtol(argv[7], NULL, 0);
+	}
+
+	if (argc > 8) {
+		chans[0]->qos->rtn = strtol(argv[8], NULL, 0);
+	}
+
+	err = bt_iso_chan_bind(conns, 1, chans);
+	if (err) {
+		shell_error(shell, "Unable to bind (err %d)", err);
+		return 0;
+	}
+
+	shell_print(shell, "ISO Channel bound");
+
+	return 0;
+}
+
+static int cmd_connect(const struct shell *shell, size_t argc, char *argv[])
+{
+	struct bt_iso_chan *chans[1];
+	int err;
+
+	if (!iso_chan.conn) {
+		shell_error(shell, "Not binded");
+		return 0;
+	}
+
+	chans[0] = &iso_chan;
+
+	err = bt_iso_chan_connect(chans, 1);
+	if (err) {
+		shell_error(shell, "Unable to connect (err %d)", err);
+		return 0;
+	}
+
+	shell_print(shell, "ISO Connect pending...");
+
+	return 0;
+}
+
+#define DATA_MTU CONFIG_BT_ISO_TX_MTU
+NET_BUF_POOL_FIXED_DEFINE(tx_pool, 1, DATA_MTU, NULL);
+
+static int cmd_send(const struct shell *shell, size_t argc, char *argv[])
+{
+	static uint8_t buf_data[DATA_MTU] = { [0 ... (DATA_MTU - 1)] = 0xff };
+	int ret, len, count = 1;
+	struct net_buf *buf;
+
+	if (argc > 1) {
+		count = strtoul(argv[1], NULL, 10);
+	}
+
+	if (!iso_chan.conn) {
+		shell_error(shell, "Not binded");
+		return 0;
+	}
+
+	len = MIN(iso_chan.qos->sdu, DATA_MTU - BT_ISO_CHAN_SEND_RESERVE);
+
+	while (count--) {
+		buf = net_buf_alloc(&tx_pool, K_FOREVER);
+		net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
+
+		net_buf_add_mem(buf, buf_data, len);
+		ret = bt_iso_chan_send(&iso_chan, buf);
+		if (ret < 0) {
+			shell_print(shell, "Unable to send: %d", -ret);
+			net_buf_unref(buf);
+			return -ENOEXEC;
```

Comment:
Strange error. Why not return ret?

Also since this is a command, don't you need to return 0?

I don't know much about Zephyr though

Reply:
See the shell for this, not to be rude or anything but it is more productive to only comment on things you actually have knowlodge that it might be a problem, or perhaps look for other shell modules such as bt.c where it basically does the same thing regarding the use of -ENOEXEC to indicate the command could not be performed.